cpp/src/arrow/gpu/cuda_memory.cc

```diff
@@ -501,5 +502,27 @@ Result<std::shared_ptr<MemoryManager>> DefaultMemoryMapper(ArrowDeviceType devic
   }
 }
 
+Result<std::shared_ptr<MemoryManager>> DefaultGPUMemoryMapper(int64_t device_id) {
```

Comment:
Two things:
1. Put it in the anonymous namespace to avoid exposing it
2. Also, rename it to `DefaultCudaMemoryMapper`.